Rover: Added logging of the guided commands

gmorph opened the pull request.

rmackay9 commented:
I guess the intention was to make the GUID message the same format as for plane and copter?  I can see wanting to do that but I wonder if it's helpful.  It seems like what's actually logged is completely different.  So for example, in the GUID's pX column it looks like either lateral acceleration or an angle error.  I guess those are completely different things?  it wouldn't make sense to compare them right?  My feeling is that it would be better to either have two different messages or different columns, some of which are filled in and some are not.

muramura commented on APMrover2/defines.h:
```diff
@@ -59,6 +59,7 @@ enum GuidedMode {
 #define LOG_SONAR_MSG 		    0x07
 #define LOG_ARM_DISARM_MSG      0x08
 #define LOG_STEERING_MSG        0x0D
+#define LOG_GUIDEDTARGET_MSG    0x0E
```
How about grouping these definitions into enum values?

gmorph replied:
We would need to do that for all vehicle types as I'd like to keep the code as consistent across all vehicles as possible.  As I'm trying to get Rover released I'll skip this for the moment but if you wanted to submit a PR with that change I think that would be fine.

gmorph commented:
Thanks for reviewing Randy.
Plane AFAIK doesn't currently log the guided commands so I was just using Copter as a template.  I had a look at each of the calls to Log_Write_GuidedTarget here:
https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/control_guided.cpp#L199
https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/control_guided.cpp#L231
https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/control_guided.cpp#L248
https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/control_guided.cpp#L265
https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/control_guided.cpp#L293
And the also use the logging structure very differently some logging destination but no velocity, others velocity but no destination, and the last logging vehicle attitude and climb rate.  Based on this I thought having a generic structure the can support logging whatever was required for GUIDED mode was the go.  In otherwords, whenever looking into the dataflash logs you know that the GUID msg always contains the GUIDED mode information and then you just need to check the code to see what is being logged in that specific GUIDED mode.  Rover especially has lots of logging messages are re-used plane/copter ones where we put the information in as best we can.  For Rover you always need to check to the code to see what is being logged in each message.

However, for the original GPS "follow me" guided mode even though I added logging I have re-considered and thought we should be logging the GPS location that the rover is being guided too rather then the lateral acceleration so I have changed that over.
Thanks, Grant.

rmackay9 commented:
As posted in skype, personally I wouldn't try to align with Plane/Copter if it makes things more difficult for Rover or the user.  So I'd vote for logging what's meaningful for Rover.  Still, it's your baby so I leave it to your best judgement.  I don't have super strong feelings about it.

gmorph commented:
Thanks Randy.  I like the flexibility of the current solution and its going to be dead simple to change later on should we decide we need to.  I'm going to leave it as is for now as I think the consistency argument is a good one.
